Move user-defined passes after SpecPropPass

lucylq opened the pull request:
Summary:
1. Use `ExecutorchBackendConfig` instead of inserting the pass.
2. Move SpecPropPass after user-defined passes. It was originally swapped here: D80840528. cc ethansfng, who has a pass that should run before SpecPropPass

Differential Revision: D87576772

JacobSzwejbka reviewed (approved):
Review automatically exported from Phabricator review in Meta.

lucylq committed:
Move user-defined passes after SpecPropPass (#15932)

Summary:

1. Use `ExecutorchBackendConfig` instead of inserting the pass.
2. Move SpecPropPass after user-defined passes. It was originally swapped here: D80840528. cc ethansfng, who has a pass that should run before SpecPropPass

Reviewed By: JacobSzwejbka, larryliu0820

Differential Revision: D87576772
